core/src/main/java/org/opensearch/sql/executor/execution/ContinuePaginatedPlan.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.executor.execution;
+
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.executor.PaginatedPlanCache;
+import org.opensearch.sql.executor.QueryId;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+
+public class ContinuePaginatedPlan extends AbstractPlan {
+
+  public static final ContinuePaginatedPlan None
+      = new ContinuePaginatedPlan(QueryId.None, "", null,
+      null, null);
+  private final String cursor;
+  private final PaginatedQueryService queryService;
+  private final PaginatedPlanCache paginatedPlanCache;
+
+  private final ResponseListener<ExecutionEngine.QueryResponse> queryResponseListener;
+
+
+  /**
+   * Create an abstract plan that can continue paginating a given cursor.
+   */
+  public ContinuePaginatedPlan(QueryId queryId, String cursor, PaginatedQueryService queryService,
+                               PaginatedPlanCache ppc,
+                               ResponseListener<ExecutionEngine.QueryResponse>
+                                   queryResponseListener) {
+    super(queryId);
+    this.cursor = cursor;
+    this.paginatedPlanCache = ppc;
+    this.queryService = queryService;
+    this.queryResponseListener = queryResponseListener;
+  }
+
+  @Override
+  public void execute() {
+    try {
+      PhysicalPlan plan = paginatedPlanCache.convertToPlan(cursor);
+      queryService.executePlan(plan, queryResponseListener);
+    } catch (Exception e) {
+      queryResponseListener.onFailure(e);
+    }
+  }
+
+  @Override
+  // TODO why can't use listener given in the constructor?
+  public void explain(ResponseListener<ExecutionEngine.ExplainResponse> listener) {
+    listener.onFailure(new UnsupportedOperationException(
```

Comment:
Probably not, this `explain` is called on cursor request only (2+ pages) and message offers to use `explain` on the first page.
But it is possible to implement, because all required plans a completely decoded from the cursor.